services/retention/service.go

```diff
@@ -79,6 +79,12 @@ func (s *Service) run() {
 			return
 
 		case <-ticker.C:
+			select {
```

Comment:
Why did you add this here?

Reply:
One of the tests set the check interval really low to 1 millisecond. This meant that, even if the service was done, the timer kept getting set off and Go kept choosing this branch so it never exited. This goes and ensures that the done channel gets priority even if there is a pending check interval.

Reply:
@jsternberg The timer shouldn't have kept getting set off. When the service is closed then the `<-s.done` case will be available. When two competing cases are available they're chosen uniformly randomly. I doubt it would have taken more than a few `ms` for the `<-s.done` case to be chosen.

I don't think adding this logic to the service for the sake of a test is the right way to go. What problems were you having with the test (I wrote it)—maybe we can refactor it?

Reply:
Ok. I'll try to figure out why it wasn't finishing and revert this back to the original code.

Even then, I'm not sure we should bother with choosing randomly. The intention appears to be that if the service is done, the service is done. Go doesn't seem to have a very good pattern for "prefer this channel over others".